Automattic/vip-go-mu-plugins
# Search: Do healthcheck on inactive indexes and build new index only for active indexes

## Description
This PR does two things:
- reverts #2810 which made health check only on active indexes
- builds new index to meet shard requirements only for active indexes

## Changelog Description

### Plugin Updated: Enterprise Search

1) Healthcheck will fire on all indexes regardless of active state. 
2) To meet shard requirements, new indexes will only be automatically built for active indexes.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test
Scenario 1: Check if healthcheck fires for all indexes regardless of state
1) Add some "bad" settings to current index "1", e.g.
```
$bad_settings = [ 'index.number_of_replicas' => 8, 'index.max_result_window' => 50 ];
$indexable = \ElasticPress\Indexables::factory()->get( 'post' );
$indexable->update_index_settings( $bad_settings );
```
2) Create a new post index version "2" `wp vip-search index-versions add post`
3) Swap to the new index "2" `wp vip-search index-versions activate post next`
4) Ensure that the old index "1" isn't returned from `get_index_settings_health_for_all_indexables()`

Scenario 2: Build new index requirements only for active index
1) Fake index shard requirement mismatch:

```
add_filter( 'ep_post_mapping', function( $mapping ) {
    $mapping['settings']['index.number_of_shards'] = 3;
return $mapping;
}, 20 );
```

2) Shell in and kick off a health check:
```
$es = new \Automattic\VIP\Search\Search();
$es->init();
$hj = new \Automattic\VIP\Search\SettingsHealthJob( $es );
$hj->check_settings_health();
```

(Can also throw a dump before `maybe_process_build()` to ensure that `$result['index_version']` matches the active index version)
